### PR TITLE
Private eth1 network

### LINKED ISSUE #142: Setup eth1 deposit contract and verify depositing works
- docker-compose with geth node and script to deploy eth1 deposit contract

### PR DESCRIPTION
**Pull request by mpetrunic**
- docker compose with geth dev node and deposit contract deployment
- refactoring of code to support different envs
- generated deposit tx works

resolves #142

### CODE REVIEW

File: src/renderer/services/deposit/networks.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const isLocal = process.env.NODE_ENV !== "production";
+
+const networks: INetworkConfig[] = [
+    {
+        networkName: "Dummy",
+        networkId: 0,
+        contract: {
+            address: "0x00000000000001",
+            depositAmount: 32,
```
Comment (morrigan) [inline src/renderer/services/deposit/networks.ts]: should this be in WEI? Prysm has 3.2 ETH for example so will be easier to handle decimals immeaditely

File: src/renderer/services/deposit/utils.ts [hunk 2]
```diff
@@ -52,7 +55,7 @@ export function generateDeposit(signingKey: KeyPair, withdrawalPubKey: BLSPubKey
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const amount: BN = new BN(
         // remove decimal zeroes
-        utils.formatUnits(utils.parseEther(depositAmount), "gwei").split(".")[0]
+        utils.formatUnits(utils.parseEther(depositAmount.toString()), "gwei").split(".")[0]
```
Comment (morrigan) [inline src/renderer/services/deposit/utils.ts]: gwei?

File: src/renderer/services/deposit/deposit-tx.ts [hunk 3]
```diff
@@ -44,14 +46,15 @@ export class DepositTx implements ITx{
         return new DepositTx(
             depositFunctionEncoded,
             depositContractAddress,
-            utils.parseEther(depositAmount).toHexString()
+            utils.parseEther(depositAmount.toString()).toHexString()
```
Comment (morrigan) [inline src/renderer/services/deposit/deposit-tx.ts]: yep, this is 32 wei only in private network case

File: test/e2e/redux/actions/deposit.action.spec.ts [hunk 4]
```diff
@@ -51,31 +53,38 @@ describe("deposit actions", () => {
...
+            keyPair,
+            Buffer.from(publicKeyStr, "hex"),
+            networkConfig.contract.depositAmount
+        );
         const depositTx = DepositTx.generateDepositTx(
             depositData, 
             networkConfig.contract.address, 
-            networkConfig.config, 
+            networkConfig.eth2Config,
             DEPOSIT_AMOUNT);
```
Comment (morrigan) [inline test/e2e/redux/actions/deposit.action.spec.ts]: redundant constant that should be removed?

File: src/renderer/services/deposit/networks.ts (hunk 1 above)
Comment (mpetrunic) [reply src/renderer/services/deposit/networks.ts]: Can't we just write 3.2? haha

File: src/renderer/services/deposit/utils.ts (hunk 2 above)
Comment (mpetrunic) [reply src/renderer/services/deposit/utils.ts]: yes, deposit contract and eth2 handles everything in gweis

File: src/renderer/services/deposit/networks.ts (hunk 1 above)
Comment (morrigan) [reply src/renderer/services/deposit/networks.ts]: I guess it's supported by the current code

**Pull request merged**